test/distributed/test_ddp_with_rpc.py

```diff
@@ -0,0 +1,160 @@
+
+
+def _call_method(method, rref, *args, **kwargs):
+    return method(rref.local_value(), *args, **kwargs)
+
+
+def _remote_method(method, rref, *args, **kwargs):
+    args = [method, rref] + list(args)
+    return rpc.rpc_sync(rref.owner(), _call_method, args=args, kwargs=kwargs)
+
+
+class SimpleNet(nn.Module):
+    def __init__(self, d_in, d_out):
+        super(SimpleNet, self).__init__()
+        self.net = nn.Linear(d_in, d_out)
+        self.relu = nn.ReLU()
+
+    def forward(self, input):
+        print("Running simple net: {input}")
+        return self.relu(self.net(input))
+
+
+class DdpModelWithRpc(nn.Module):
+    def __init__(self, remote_server, process_group_name):
+        super(DdpModelWithRpc, self).__init__()
+        self.net1 = DDP(SimpleNet(5, 8), process_group=process_group_name)
+        self.rref = rpc.remote(remote_server, SimpleNet, 8, 5)
+        self.net2 = DDP(SimpleNet(5, 3), process_group=process_group_name)
+
+    def forward(self, x):
+        x = self.net1(x)
+        x = _remote_method(SimpleNet.forward, self.rref, x)
+        return self.net2(x)
+
+
+class TestDdpWithRpc(TestCase):
+    TRAINER_NAME_TEMPLATE = "trainer:{:02d}"
+    REMOTE_WORKER_NAME = "remote_worker"
+    TRAINER_GROUP = "trainer_group"
+    NUM_TRAINERS = 3
+
+    def setUp(self):
+        super(TestDdpWithRpc, self).setUp()
+        self.processes = []
+        os.environ["MASTER_ADDR"] = str(MASTER_ADDR)
+        os.environ["MASTER_PORT"] = str(MASTER_PORT)
+        os.environ["WORLD_SIZE"] = str(self.NUM_TRAINERS + 1)
+
+    def tearDown(self):
+        super(TestDdpWithRpc, self).tearDown()
+        for p in self.processes:
+            p.terminate()
+
+    def join_processes(self):
+        for p in self.processes:
+            p.join()
+
+    def run_one_mini_batch(self):
+        with dist_autograd.context() as context_id:
+            # Forward pass (create references on remote nodes).
+            rref1 = rpc.remote(dst_name, random_tensor)
+            rref2 = rpc.remote(dst_name, random_tensor)
+            loss = rref1.to_here() + rref2.to_here()
+
+            # Backward pass (run distributed autograd).
+            dist_autograd.backward([loss.sum()])
+
+            # Build DistributedOptimizer.
+            dist_optim = DistributedOptimizer(
+                optim.SGD, [rref1, rref2], lr=0.05
+            )
+
+            # Run the distributed optimizer step.
+            dist_optim.step()
+
+    def run_trainer(self, rank, func):
+        trainer_name = self.TRAINER_NAME_TEMPLATE.format(rank)
+        print(f"Starting trainer: {trainer_name}...")
+        #  For DDP communications
+        dist.init_process_group(
+            # TODO: test other rpc backend.
+            "gloo",
+            group_name=self.TRAINER_GROUP,
+            rank=rank,
+            world_size=self.NUM_TRAINERS,
+        )
+        # This group includes the remote worker
+        rpc.init_rpc(
```

Comment:
This `init_rpc` will fail because the master port is already in use by `init_process_group()` call above.